### PR TITLE
Feature/decorate class

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
## Feature/decorate class
In this app functionality improvements; 

#### Prepare base Decorator
- Make sure that it inherits from Namable.
- In the constructor assign a namable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name(name)
+    name.length >= 10 ? name[0..9].capitalize : name.capitalize
```
Comment (omar-labana) [inline nameable.rb]: - [x] This method should raise a NotImplementedError according to the project requirements. Kindly update it.

File: person.rb [hunk 2]
```diff
@@ -1,19 +1,34 @@
+require_relative './nameable'
```
Comment (omar-labana) [inline person.rb]: - [x] According to the project requirements, your Person class should inherit from the Nameable class, please update it.

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @blessedjasonmwanza, 

Excellent work so far! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
- [x]  According to the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_decorator_v2.md), you should prepare a `base Decorator` then extend two classes from it, kindly take another look at the project's requirements to see what is missing.
- [x] _Please, check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: nameable.rb (hunk 1 above)
Comment (blessedjasonmwanza) [reply nameable.rb]: Actioned 🤓  -> Thanks a lot for sporting this.

File: person.rb (hunk 2 above)
Comment (blessedjasonmwanza) [reply person.rb]: Thanks a million, @omar-labana  for clarity - Fixed the issue as per the guide.

**Pull request merged**